conda_lock/pypi_solver.py

```diff
@@ -0,0 +1,286 @@
+from poetry.repositories.repository import Repository
+from poetry.utils.env import Env
+
+from conda_lock import src_parser
+from conda_lock.src_parser.pyproject_toml import get_lookup as get_forward_lookup
+from conda_lock.src_parser.pyproject_toml import normalize_pypi_name
+
+
+class PlatformEnv(Env):
+    def __init__(self, python_version, platform):
+        super().__init__(path=Path(sys.prefix))
+        if platform == "linux-64":
+            # FIXME: in principle these depend on the glibc in the conda env
+            self._platforms = ["manylinux_2_17_x86_64", "manylinux2014_x86_64"]
+        elif platform == "osx-64":
+            self._platforms = ["macosx_10_9_x86_64"]
+        elif platform == "osx-arm64":
+            self._platforms = [
+                "macosx_11_0_arm64",
+                "macosx_11_0_universal2",
+                "macosx_10_16_universal2",
+                "macosx_10_15_universal2",
+                "macosx_10_14_universal2",
+                "macosx_10_13_universal2",
+                "macosx_10_12_universal2",
+                "macosx_10_11_universal2",
+                "macosx_10_10_universal2",
+                "macosx_10_9_universal2",
+                "macosx_10_8_universal2",
+                "macosx_10_7_universal2",
+                "macosx_10_6_universal2",
+                "macosx_10_5_universal2",
+                "macosx_10_4_universal2",
+            ]
+        elif platform == "win-64":
+            self._platforms = ["win_amd64"]
+        else:
+            raise ValueError(f"Unsupported platform '{platform}'")
+        self._python_version = tuple(map(int, python_version.split(".")))
+
+        if platform.startswith("osx-"):
+            self._sys_platform = "darwin"
+        elif platform.startswith("linux-"):
+            self._sys_platform = "linux"
+        elif platform.startswith("win-"):
+            self._sys_platform = "win32"
+        else:
+            raise ValueError(f"Unsupported platform '{platform}'")
+
+    def get_supported_tags(self):
+        """
+        Mimic the output of packaging.tags.sys_tags() on the given platform
+        """
+        return list(
+            cpython_tags(python_version=self._python_version, platforms=self._platforms)
+        ) + list(
+            compatible_tags(
+                python_version=self._python_version, platforms=self._platforms
+            )
+        )
+
+    def get_marker_env(self):
+        return {
+            # "implementation_name": implementation_name,
+            # "implementation_version": iver,
+            # "os_name": os.name,
+            # "platform_machine": platform.machine(),
+            # "platform_release": platform.release(),
+            # "platform_system": platform.system(),
+            # "platform_version": platform.version(),
+            "python_full_version": ".".join([str(c) for c in self._python_version]),
+            # "platform_python_implementation": platform.python_implementation(),
```

Comment:
We can probably pull that out too from our python version information (at least for conda-forge)

Reply:
Right, these are the ones you tend to see in the wild. More can be added if the need arises.